testhelper/docker/resource/transformer/transformer.go

```diff
@@ -21,12 +26,62 @@ type config struct {
 	repository   string
 	tag          string
 	exposedPorts []string
-	env          []string
+	envs         []string
+	extraHosts   []string
+}
+
+func (c *config) updateBackendConfigURL(url string) {
+	found := false
+	for i, env := range c.envs {
+		if strings.HasPrefix(env, "CONFIG_BACKEND_URL=") {
+			found = true
+			c.envs[i] = fmt.Sprintf("CONFIG_BACKEND_URL=%s", url)
+		}
+	}
+	if !found {
+		c.envs = append(c.envs, fmt.Sprintf("CONFIG_BACKEND_URL=%s", url))
+	}
+}
+
+// WithTransformations will mock BE config to set transformation for given transformation versionID to transformation function map
+//
+// - events with transformationVersionID not present in map will not be transformed and transformer will return 404 for those requests
+//
+// - WithTransformations should not be used with WithConfigBackendURL option
+//
+// - only javascript transformation functions are supported
+//
+// e.g.
+//
+//	WithTransformations(map[string]string{
+//				"transform-version-id-1": `export function transformEvent(event, metadata) {
+//											event.transformed=true
+//											return event;
+//										}`,
+//			})
+func WithTransformations(transformations map[string]string, cleaner resource.Cleaner) func(*config) {
+	return func(conf *config) {
+		mux := http.NewServeMux()
+		mockBackendConfigServer := &mockHttpServer{Transformations: transformations}
+		mux.HandleFunc(getByVersionIdEndPoint, mockBackendConfigServer.handleGetByVersionId)
+		backendConfigSvc := kithttptest.NewServer(mux)
```

Comment:
creation of the httptest server shouldn't be the responsibility of this function
```suggestion
		backendConfigSvc := NewTestBackendConfigServer(transformations)
```